packages/graphql/src/schema/get-obj-field-meta.ts

```diff
@@ -318,6 +315,17 @@ function getObjFieldMeta({
                     kind: "Enum",
                     ...baseField,
                 };
+
+                if (defaultDirective) {
+                    const value = defaultDirective.arguments?.find((a) => a.name.value === "value")?.value;
+
+                    if (value?.kind !== Kind.ENUM) {
+                        throw new Error("@default value on enum fields must be an enum value");
+                    }
+
+                    enumField.defaultValue = (value as EnumValueNode).value;
+                }
```

Comment:
**suggestion (blocking)**: We could improve these typings a bit, to avoid casting. By using a helper function to check the value kind:
```typescript
function isEnumValue(value: ValueNode): value is EnumValueNode {
     return value.kind === Kind.ENUM;
}
```
This function, for the time being, could be in the same file at the bottom.

This way,  this check can be safely typed:

```suggestion
                if (defaultDirective) {
                    const defaultValue = defaultDirective.arguments?.find((a) => a.name.value === "value")?.value;

                    if (!defaultValue || !isEnumValue(defaultValue)) {
                        throw new Error("@default value on enum fields must be an enum value");
                    }

                    enumField.defaultValue = defaultValue.value;
                }
```